src/Bicep.Core/Text/SpellChecker.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Bicep.Core.Text
+{
+    public static class SpellChecker
+    {
+        public static string? GetSpellingSuggestion(string name, IEnumerable<string> candidates)

```

Comment:
I basically stole the optimizations from the TypeScript source code :p. The credits goes to the TypeScript team.

Reply:
Btw, how does this deal with casing mismatches?

Reply:
If a candidate is the name as the name except for case, that candidate is returned immediately (see `nameUpperCase.Equals(candidateUpperCase)` below), and no edit distance will be calculated.

Reply:
Nice!